mbed_targets/mbed_targets.py

```diff
@@ -84,22 +88,71 @@ class DatabaseMode(Enum):
 
     OFFLINE = 0
     ONLINE = 1
+    AUTO = 2
+
+
+def get_target(product_code: str, mode: DatabaseMode) -> MbedTarget:
```

Comment:
Is `product_code` always `hex`? I wonder if we should take that into account in the representation of `MbedTargets` and when searching for target by it's `product_code`. I.e.: if we're just comparing strings, a `hex` product code of `0x123` which is equivalent to `0x0123` and `0123` won't match. While I'm skeptic about polluting `mbed-targets` with that knowledge, I'm wondering where the fact that product codes are of certain format should live and be documented.